packages/xrpl/src/models/transactions/common.ts

```diff
@@ -696,3 +733,170 @@ const _DOMAIN_ID_LENGTH = 64
 export function isDomainID(domainID: unknown): domainID is string {
   return isString(domainID) && domainID.length === _DOMAIN_ID_LENGTH
 }
+
+/* eslint-disable max-lines-per-function -- Required here as structure validation is verbose. */
+/* eslint-disable max-statements -- Required here as structure validation is verbose. */
+
+/**
+ * Validates if MPTokenMetadata adheres to XLS-89d standard.
+ *
+ * @param input - Hex encoded MPTokenMetadata.
+ * @returns Validation error messages and boolean denoting if MPTokenMetadata adheres to XLS-89d standard.
+ */
+export function validateMPTokenMetadata(input: string): {
+  isValid: boolean
+  validationMessages: string[]
+} {
+  const validationMessages: string[] = []
+
+  if (!isHex(input) || input.length / 2 > MAX_MPT_META_BYTE_LENGTH) {
+    validationMessages.push(
+      `MPTokenMetadata must be in hex format and max ${MAX_MPT_META_BYTE_LENGTH} bytes.`,
+    )
+    return { validationMessages, isValid: false }
+  }
+
+  let jsonMetaData: unknown
+
+  try {
+    jsonMetaData = JSON.parse(hexToString(input))
+  } catch (err) {
+    validationMessages.push(
+      `MPTokenMetadata is not properly formatted as JSON - ${String(err)}`,
+    )
+    return { validationMessages, isValid: false }
+  }
+
+  if (
+    !(
+      jsonMetaData != null &&
+      typeof jsonMetaData === 'object' &&
+      !Array.isArray(jsonMetaData)
+    )
+  ) {
+    validationMessages.push(
+      'MPTokenMetadata is not properly formatted as per XLS-89d.',
+    )
+    return { validationMessages, isValid: false }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- It must be some JSON object.
+  const obj = jsonMetaData as Record<string, unknown>
+
+  // validating structure
+  const incorrectRequiredFields = MPT_META_REQUIRED_FIELDS.filter(
+    (field) => !isString(obj[field]),
+  )
+
+  if (incorrectRequiredFields.length > 0) {
+    incorrectRequiredFields.forEach((field) =>
+      validationMessages.push(`${field} is required and must be string.`),
+    )
+    return { validationMessages, isValid: false }
+  }
+
+  if (obj.desc != null && !isString(obj.desc)) {
+    validationMessages.push(`desc must be a string.`)
+    return { validationMessages, isValid: false }
+  }
+
+  if (obj.asset_subclass != null && !isString(obj.asset_subclass)) {
+    validationMessages.push(`asset_subclass must be a string.`)
+    return { validationMessages, isValid: false }
+  }
+
+  if (
+    obj.additional_info != null &&
+    !(
+      isString(obj.additional_info) ||
+      (typeof obj.additional_info === 'object' &&
+        !Array.isArray(obj.additional_info))
+    )
```

Comment:
```suggestion
    obj.additional_info != null &&
    !isString(obj.additional_info) &&
    (
      typeof obj.additional_info !== 'object' ||
      Array.isArray(obj.additional_info) ||
      obj.additional_info === null // include if you want to disallow null as object
    )
```
Nested negation is difficult to read.